src/motion/functionality/types.ts

```diff
@@ -10,15 +10,23 @@ export interface FunctionalProps extends MotionProps {
 }
 
 export interface FunctionalComponentDefinition {
+    key: string
     shouldRender: (props: MotionProps) => boolean
-    component: ComponentType<FunctionalProps>
+    Component: ComponentType<FunctionalProps>
```

Comment:
Is `Functional` the right terminology here? It is a bout the functionality, right? Maybe we can come up with better naming.

Reply:
Yeah I feel like it could do with a better name - at some point I want to open this API up so that would be a good time to come back to this.